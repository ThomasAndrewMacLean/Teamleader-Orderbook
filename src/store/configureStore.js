import { createStore, applyMiddleware, compose } from 'redux';
import rootReducer from './../reducers/reducers';
import thunk from 'redux-thunk';
import { createLogger } from 'redux-logger';
import { routerMiddleware } from 'react-router-redux';

import createHistory from 'history/createBrowserHistory';



const logger = createLogger({
    collapsed: true
});

const history = createHistory();
const routerMw = routerMiddleware(history);

const composeEnhancers =
    typeof window === 'object' &&
        window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ ?
        window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__({
            // Specify extensionâ€™s options like name, actionsBlacklist, actionsCreators, serialize...
        }) : compose;


export default function configureStore() {
    return createStore(

        rootReducer,
        composeEnhancers(
            applyMiddleware(routerMw, logger, thunk))
    );
}
