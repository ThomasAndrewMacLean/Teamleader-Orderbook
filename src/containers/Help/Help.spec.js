import React from 'react';
import ReactDOM from 'react-dom';
import ConnectedHelp, { Help } from './Help';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { shallow } from 'enzyme';
import * as mocks from './../../test-helpers/mocks';

const mockStore = configureStore();

let initialState;
let store;
let wrapper;
const mockDeleteProduct = jest.fn();
const mockAddQuantity = jest.fn();
beforeEach(() => {
    jest.clearAllMocks();
    Help.prototype._slider = { addEventListener };

    initialState = {
        data: {
            orders: mocks.getOrders(),
            products: mocks.getProducts(),
            customers: mocks.getCustomers()
        }
    };
    //creates the store with any initial state or middleware needed  
    store = mockStore(initialState);
    // wrapper = mount(<Provider store={store}><ConnectedList /></Provider>);
    wrapper = shallow(<Help order={mocks.getOrders()[0]} item={mocks.getOrders()[0].items[0]} products={mocks.getProducts()} addQuantity={mockAddQuantity} deleteProduct={mockDeleteProduct} />);
});

it('snapshot', () => {
    const tree = wrapper;
    expect(tree).toMatchSnapshot();
});

it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Provider store={store}><ConnectedHelp /></Provider>, div);
    ReactDOM.unmountComponentAtNode(div);
});
it('renders HELP titel in header', () => {
    expect(wrapper.find('.App-header').text()).toContain('HELP');
});

it('renders back button in header', () => {
    expect(wrapper.find('.App-header').text()).toContain('BACK');
});

it('renders list of helpitems', () => {
    expect(wrapper.find('ul')).toHaveLength(1);
});
it('renders list of helpitems with class info-list', () => {
    expect(wrapper.find('ul').hasClass('info-list')).toBeTruthy();
});
it('back button calls function goHome', () => {
    const mock = jest.fn();
    let history = [];
    const mockHistory = { push: (x) => mock(history.push(x)) };

    wrapper = shallow(<Help history={mockHistory} order={mocks.getOrders()[0]} products={mocks.getProducts()} customers={mocks.getCustomers()} />);

    const spy = jest.spyOn(Help.prototype, 'goHome');
    wrapper.find('#back-button').simulate('click');
    expect(spy).toHaveBeenCalledTimes(1);
});

it('back button pushes / on history', () => {
    const mock = jest.fn();
    let history = [];
    const mockHistory = { push: (x) => mock(history.push(x)) };

    wrapper = shallow(<Help history={mockHistory} order={mocks.getOrders()[0]} products={mocks.getProducts()} customers={mocks.getCustomers()} />);

    wrapper.find('#back-button').simulate('click');
    expect(history).toHaveLength(1);
    expect(history[0]).toBe('/');
});