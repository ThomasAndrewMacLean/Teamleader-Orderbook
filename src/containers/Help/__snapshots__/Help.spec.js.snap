// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Help
    addQuantity={[Function]}
    deleteProduct={[Function]}
    item={
        Object {
            "product-id": "B102",
            "quantity": "10",
            "total": "49.90",
            "unit-price": "4.99",
          }
    }
    order={
        Object {
            "customer-id": "1",
            "id": "1",
            "items": Array [
              Object {
                "product-id": "B102",
                "quantity": "10",
                "total": "49.90",
                "unit-price": "4.99",
              },
            ],
            "total": "49.90",
          }
    }
    products={
        Array [
            Object {
              "category": "1",
              "description": "Screwdriver",
              "id": "A101",
              "price": "9.75",
            },
            Object {
              "category": "1",
              "description": "Electric screwdriver",
              "id": "A102",
              "price": "49.50",
            },
            Object {
              "category": "2",
              "description": "Basic on-off switch",
              "id": "B101",
              "price": "4.99",
            },
            Object {
              "category": "2",
              "description": "Press button",
              "id": "B102",
              "price": "4.99",
            },
            Object {
              "category": "2",
              "description": "Switch with motion detector",
              "id": "B103",
              "price": "12.95",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <header
          className="App-header"
>
          <div
                    className="quatre-col header"
                    id="back-button"
                    onClick={[Function]}
          >
                    BACK
          </div>
          <div
                    className="three-quatre-col header"
          >
                    <p
                              className="pullright"
                    >
                              HELP
                    </p>
          </div>
</header>,
        <ul
          className="info-list"
>
          <li>
                    Click on details to see the individual products of an order.
          </li>
          <li>
                    On mobile you can delete a product by pressing on it. Not-mobile devices will see a cross on hover. Both will still first ask for confirmation.
          </li>
          <li>
                    On every change the price is checked to see if you get any discounts.
          </li>
          <li>
                    You can swipe back to the main overview by swiping from left to right.
          </li>
          <li>
                    Extra orders can be placed by clicking on the + button on the order overview page.
          </li>
          <li>
                    As long as an order has not been placed you can add extra products. Click the button at the bottom left to choose an extra product. By default one will be added.
          </li>
</ul>,
      ],
      "className": "App",
    },
    "ref": [Function],
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="quatre-col header"
              id="back-button"
              onClick={[Function]}
>
              BACK
</div>,
            <div
              className="three-quatre-col header"
>
              <p
                            className="pullright"
              >
                            HELP
              </p>
</div>,
          ],
          "className": "App-header",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "BACK",
              "className": "quatre-col header",
              "id": "back-button",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "BACK",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <p
                className="pullright"
>
                HELP
</p>,
              "className": "three-quatre-col header",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "HELP",
                "className": "pullright",
              },
              "ref": null,
              "rendered": "HELP",
              "type": "p",
            },
            "type": "div",
          },
        ],
        "type": "header",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li>
              Click on details to see the individual products of an order.
</li>,
            <li>
              On mobile you can delete a product by pressing on it. Not-mobile devices will see a cross on hover. Both will still first ask for confirmation.
</li>,
            <li>
              On every change the price is checked to see if you get any discounts.
</li>,
            <li>
              You can swipe back to the main overview by swiping from left to right.
</li>,
            <li>
              Extra orders can be placed by clicking on the + button on the order overview page.
</li>,
            <li>
              As long as an order has not been placed you can add extra products. Click the button at the bottom left to choose an extra product. By default one will be added.
</li>,
          ],
          "className": "info-list",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Click on details to see the individual products of an order.",
            },
            "ref": null,
            "rendered": "Click on details to see the individual products of an order.",
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "On mobile you can delete a product by pressing on it. Not-mobile devices will see a cross on hover. Both will still first ask for confirmation.",
            },
            "ref": null,
            "rendered": "On mobile you can delete a product by pressing on it. Not-mobile devices will see a cross on hover. Both will still first ask for confirmation.",
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "On every change the price is checked to see if you get any discounts.",
            },
            "ref": null,
            "rendered": "On every change the price is checked to see if you get any discounts.",
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "You can swipe back to the main overview by swiping from left to right.",
            },
            "ref": null,
            "rendered": "You can swipe back to the main overview by swiping from left to right.",
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Extra orders can be placed by clicking on the + button on the order overview page.",
            },
            "ref": null,
            "rendered": "Extra orders can be placed by clicking on the + button on the order overview page.",
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "As long as an order has not been placed you can add extra products. Click the button at the bottom left to choose an extra product. By default one will be added.",
            },
            "ref": null,
            "rendered": "As long as an order has not been placed you can add extra products. Click the button at the bottom left to choose an extra product. By default one will be added.",
            "type": "li",
          },
        ],
        "type": "ul",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <header
            className="App-header"
>
            <div
                        className="quatre-col header"
                        id="back-button"
                        onClick={[Function]}
            >
                        BACK
            </div>
            <div
                        className="three-quatre-col header"
            >
                        <p
                                    className="pullright"
                        >
                                    HELP
                        </p>
            </div>
</header>,
          <ul
            className="info-list"
>
            <li>
                        Click on details to see the individual products of an order.
            </li>
            <li>
                        On mobile you can delete a product by pressing on it. Not-mobile devices will see a cross on hover. Both will still first ask for confirmation.
            </li>
            <li>
                        On every change the price is checked to see if you get any discounts.
            </li>
            <li>
                        You can swipe back to the main overview by swiping from left to right.
            </li>
            <li>
                        Extra orders can be placed by clicking on the + button on the order overview page.
            </li>
            <li>
                        As long as an order has not been placed you can add extra products. Click the button at the bottom left to choose an extra product. By default one will be added.
            </li>
</ul>,
        ],
        "className": "App",
      },
      "ref": [Function],
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="quatre-col header"
                id="back-button"
                onClick={[Function]}
>
                BACK
</div>,
              <div
                className="three-quatre-col header"
>
                <p
                                className="pullright"
                >
                                HELP
                </p>
</div>,
            ],
            "className": "App-header",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "BACK",
                "className": "quatre-col header",
                "id": "back-button",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "BACK",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <p
                  className="pullright"
>
                  HELP
</p>,
                "className": "three-quatre-col header",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "HELP",
                  "className": "pullright",
                },
                "ref": null,
                "rendered": "HELP",
                "type": "p",
              },
              "type": "div",
            },
          ],
          "type": "header",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li>
                Click on details to see the individual products of an order.
</li>,
              <li>
                On mobile you can delete a product by pressing on it. Not-mobile devices will see a cross on hover. Both will still first ask for confirmation.
</li>,
              <li>
                On every change the price is checked to see if you get any discounts.
</li>,
              <li>
                You can swipe back to the main overview by swiping from left to right.
</li>,
              <li>
                Extra orders can be placed by clicking on the + button on the order overview page.
</li>,
              <li>
                As long as an order has not been placed you can add extra products. Click the button at the bottom left to choose an extra product. By default one will be added.
</li>,
            ],
            "className": "info-list",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Click on details to see the individual products of an order.",
              },
              "ref": null,
              "rendered": "Click on details to see the individual products of an order.",
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "On mobile you can delete a product by pressing on it. Not-mobile devices will see a cross on hover. Both will still first ask for confirmation.",
              },
              "ref": null,
              "rendered": "On mobile you can delete a product by pressing on it. Not-mobile devices will see a cross on hover. Both will still first ask for confirmation.",
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "On every change the price is checked to see if you get any discounts.",
              },
              "ref": null,
              "rendered": "On every change the price is checked to see if you get any discounts.",
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "You can swipe back to the main overview by swiping from left to right.",
              },
              "ref": null,
              "rendered": "You can swipe back to the main overview by swiping from left to right.",
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Extra orders can be placed by clicking on the + button on the order overview page.",
              },
              "ref": null,
              "rendered": "Extra orders can be placed by clicking on the + button on the order overview page.",
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "As long as an order has not been placed you can add extra products. Click the button at the bottom left to choose an extra product. By default one will be added.",
              },
              "ref": null,
              "rendered": "As long as an order has not been placed you can add extra products. Click the button at the bottom left to choose an extra product. By default one will be added.",
              "type": "li",
            },
          ],
          "type": "ul",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
